api/v1alpha1/clusterextension_types.go

```diff
@@ -122,9 +122,9 @@ func init() {
 // ClusterExtensionStatus defines the observed state of ClusterExtension
 type ClusterExtensionStatus struct {
 	// +optional
-	InstalledBundleResource string `json:"installedBundleResource,omitempty"`
+	InstalledBundle *BundleMetadata `json:"installedBundle,omitempty"`
 	// +optional
-	ResolvedBundleResource string `json:"resolvedBundleResource,omitempty"`
+	ResolvedBundle *BundleMetadata `json:"resolvedBundle,omitempty"`
```

Comment:
Can we make this change on the `ClusterExtension` API without it being considered a breaking change?

Reply:
We could if we had to. We could add this field and deprecate the old one. I don't think it's worth it to go to the trouble in an alpha API that we'll likely drop in the next couple of months.

I suppose you could argue, "why make the change at all if we're planning to remove the API?". I'd say that:
1. `ClusterExtension` is still more mature, so making this change there is essentially a preview for existing users of `ClusterExtension`
2. It's easier, at least in this case, to keep `ClusterExtension` and `Extension` functionality aligned in the areas where there is no fundamental difference between the two.

Reply:
>I don't think it's worth it to go to the trouble in an alpha API that we'll likely drop in the next couple of months

Totally fair. I just wanted to explicitly call this out to make sure that we were making this breaking change as a conscious decision since we have intentionally tried to avoid those in the past, even on our alpha APIs. 

I do think in general, we should err on the side of not making breaking changes due to potential side effect to the cluster/clients but I do understand that it should be _okay_ since we are in an alpha state both as a project and for the APIs.